src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { getArticleById } from "../api";
+import { useParams } from "react-router-dom";
+import SingleArticleCard from "./SingleArticleCard";
+
+function SingleArticle() {
+  const [article, setArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
isLoading is really false when the component mounts as data fetching does not start until the useEffect fires (after the component is first added to the DOM). You should set it as true at the start of the useEffect.

Reply:
Actually I take this back somewhat, I'd leave it as true initially - just be sure that for components that might refetch data without being unmounted that isLoading is set to true again inside any useEffect! (It doesn't really apply in this case but would for things like article/:article_id